Develop jiaxing

applepie1026 opened the pull request:
a test of pull request

applepie1026 committed:
a test of git push

applepie1026 committed:
another test

applepie1026 committed:
a push to yuxiao-qin

applepie1026 committed:
another push to yuxiaoQin

applepie1026 committed:
h2phase demo

applepie1026 committed:
wrapped phase simulation

applepie1026 committed:
the model of param_search_spcae

applepie1026 committed:
model of maximum untested

applepie1026 committed:
test push

applepie1026 committed:
a test of pushing

applepie1026 committed:
test1

applepie1026 committed:
the model of maximum coh_t and caculation of parameters

yuxiao-qin commented on scientific_research_with_python_demo/main.py:
```diff
@@ -1,34 +1,73 @@
-
 def wrap_phase(phase:np.ndarray)->np.ndarray:
-    return (phase + np.pi) % (2 * np.pi) - np.pi
+    return (phase + np.pi) % (2 * np.pi) - np.pi# 缠绕相位,测试，测试,test
```
请使用有意义的注释。

yuxiao-qin commented on scientific_research_with_python_demo/main.py:
```diff
@@ -1,34 +1,73 @@
-    noise_phase = generate_phase_noise(noise_level)
-
-    return wrap_phase(dv_phase + dh_phase + noise_phase)
+    noise_phase = generate_phase_noise(noise_level)*4*np.pi/WAVELENGTH
```
*4*np.pi/WAVELENGTH 的逻辑应该放到 generate_phase_noise() 里面。

一个函数应该包括一个完整的功能，转换到 phase 的功能是 `generate_phase_noise` 的一部分。

yuxiao-qin commented on scientific_research_with_python_demo/main.py:
```diff
@@ -1,34 +1,73 @@
     """Calculate phase difference from velocity difference (of two points).
     """
     temporal_baseline = 12  # [unit:d]
+    Btemp=temporal_baseline*time_range
```
这个名字起得不好。应该是：

```python
temporal_baseline = ...
temporal_samples = temporal_baseline * time_range
```

yuxiao-qin commented on scientific_research_with_python_demo/main.py:
```diff
@@ -1,34 +1,73 @@
 
 def h2phase(h:float)->np.ndarray:
-    pass
+    B_normal=300#[unit:m]
```
```python
normal_baseline = ... 
```

另外，注意，每幅影像的 normal baseline 实际上是不一样的，不应该是一个常数

yuxiao-qin commented on scientific_research_with_python_demo/main.py:
```diff
@@ -1,34 +1,73 @@
 def h2phase(h:float)->np.ndarray:
-    pass
+    B_normal=300#[unit:m]
+    H=780000# satellite vertical height[m]
```
“#” 应当空两格

yuxiao-qin commented on scientific_research_with_python_demo/main.py:
```diff
@@ -1,34 +1,73 @@
-    pass
+    B_normal=300#[unit:m]
+    H=780000# satellite vertical height[m]
+    theta=23*np.pi/180
```
`theta` 名字有点模糊，这里建议用 `incidence_angle`

yuxiao-qin commented on scientific_research_with_python_demo/main.py:
```diff
@@ -1,34 +1,73 @@
+    H=780000# satellite vertical height[m]
+    theta=23*np.pi/180
+    R=H/np.cos(theta)
+    Bn=4*np.pi*B_normal/(WAVELENGTH*R*np.sin(theta))
```
这个地方不应该是 Bn，应该叫 "h2ph_coefficient"，或者这个可以简称 "h2ph_coef"

yuxiao-qin commented on scientific_research_with_python_demo/main.py:
```diff
@@ -1,34 +1,73 @@
+    input: step  :step for search
+           2*Nsearch :number of paramters we can search   
+    创建参数搜索区域[-10:1:10]"""
+    Search_space=np.mat(np.arange(-Nsearch*step,Nsearch*step,step))
```
请用 link_with_underscore 的命名方法

yuxiao-qin commented on scientific_research_with_python_demo/main.py:
```diff
@@ -1,34 +1,73 @@
+    return Search_space_phase
+
+def construct_search_space(phase_search1,phase_serach2,num_serach)->np.ndarray:
+    search_space=np.kron(phase_search1,np.ones(1,num_serach[1]))+np.kron(np.ones(1,num_serach[0]),phase_serach2) 
```
这个地方我们要再讨论一下

applepie1026 committed:
1.the prograss of parameters estimation by using
best_index
2. request more information about ind2sub by python
3.tests about construct paramters search_space
and maximum are done

applepie1026 commented:
1.the prograss of parameters estimation by using best_index
2. request more information about ind2sub by python
3.tests about construct paramters search_space
and maximum are done
see the commit 
["7bb0b8672297620ce937f72849f8db302647332b"](https://github.com/yuxiao-qin/scientific_research_with_python_demo/pull/4/commits/7bb0b8672297620ce937f72849f8db302647332b)
@yuxiao-qin

applepie1026 reviewed (commented):
1.the prograss of parameters estimation by using best_index
2. request more information about ind2sub by python
3.tests about construct paramters search_space
and maximum are done

applepie1026 committed:
Main commit
1. correct the model of h2phase including the caculation of 'baseline' in the process of topography
2.tests about h2phase、v2phase、construct_simulated_arc_phase completed

applepie1026 committed:
code beautifually
documenting string、variables comments

applepie1026 committed:
code beautification
1. code comments and documents
2. nomalize the names of variables

applepie1026 committed:
test

applepie1026 committed:
code woking
1.code commets upgraded
2. create 'periodogram.py' to test the model of sim_arc_phase
3. normalize input variables such as 'time_range' ,'nomal_baselin',
v and h

applepie1026 committed:
code upgrade
1. delete module'maximum_temporal_coh'
2. append module 'sim_temporal_coh'
3. appdend module 'maximum_coh'
4. test of periodogram  at 1st iteration

applepie1026 committed:
code working:periodogram
1. append a module 'periodogram' to
run the whole project
2. test of module'periodogram'
